<?xml version="1.0"?>

<launch>
  <arg name="notes" default="No notes provided."/>
  <arg name="dry_run" default="false"/>
  <arg name="test" default="false" />

  <include file="$(find droplet_underwater_assembly)/description.launch" />

  <node pkg="nodelet" type="nodelet" name="bluerov_nodelet_manager" args="manager" output="screen" respawn="true">
      <param name="num_worker_threads" value="3" />
  </node>

  <include file="$(find droplet_underwater_assembly)/stag.launch"/>

  <group unless="$(arg test)">
    <include file="$(find droplet_underwater_assembly)/apm.launch" />

    <include file="$(find spinnaker_sdk_camera_driver)/launch/acquisition.launch">
      <arg name="nodelet_manager_name" value="bluerov_nodelet_manager"/> 
      <arg name="config_file" value="$(find droplet_underwater_assembly)/param/flir_pool_calibration_1440_1080.yaml"/>
      <arg name="start_nodelet_manager" value="false"/>
      <arg name="color" value="false"/>
      <arg name="exposure_time" value="8100"/>
    </include>

    <node pkg="rosbag" type="record" name="experiment_rosbag" args="-o /home/bluerov/catkin-workspace/install/share/droplet_underwater_assembly/static_assets/ /camera_array/cam0/camera_info /rosout /bluerov_controller/ar_tag_detector /camera_array/cam0/camera_info /tf /tf_static /mavros/rc/override /mavros/imu/data_raw /imu/data /mavros/imu/data /goal_pose"></node>

  </group>

  <node pkg="imu_complementary_filter" type="complementary_filter_node" name="complementary_filter" required="true">
      <remap from="/imu/data_raw" to="/mavros/imu/data_raw"/>
  </node>

  <node pkg="droplet_underwater_assembly" output="screen" respawn="false" type="assembly_main.py" name="assembly_main_node" required="true"> 
    <param name="dry_run" type="bool" value="$(arg dry_run)" unless="$(arg test)"/>
    <param name="dry_run" type="bool" value="true" if="$(arg test)"/>
  </node>
</launch>
